src/dvc_data/index/index.py

```diff
@@ -224,6 +241,84 @@ def exists(self, entry: "DataIndexEntry", refresh: bool = False) -> bool:
         finally:
             self.index.commit()
 
+    def _bulk_exists_from_cache(
```

Comment:
I think it's better to inline this within `bulk_exists()`.

`_from_cache` is misleading.